flybywiresim/a32nx
# feat: chrono auto brightness

Fixes #[issue_no]

## Summary of Changes


- A basic CSS-based implementation which modifies the brightness of the Chrono text depending on the time of the day. 
- This feature considers the ~~`LOCAL TIME`~~ `TIME OF DAY` simvar available by default in the sim.
- `TIME OF DAY`, type "enum": Returns 0,1,2,3 which corresponds to dawn, daylight, dusk and night respectively marked as phases of the day. Brightness change should be observed from phase 0->1 and phase 2->3.
- Works in polar regions and similar locations where days with no sunrise/sunsets exist
- This method was considered because of 2 main reasons: there is currently no way to implement the IRL chrono brightness sensor in sim (at least not known to me :P), and the text on the chrono display is not controlled by any brightness knobs either.
- The auto brightness is a fair trade-off between poor text visibility in daylight and an overly bright chrono in the night.

## Screenshots (if necessary)



**BEFORE**
![image](https://user-images.githubusercontent.com/29005005/120905828-7071a700-c672-11eb-8367-d5f49a809348.png)

**AFTER**
![image](https://user-images.githubusercontent.com/29005005/120905838-81221d00-c672-11eb-8992-844e2761e26e.png)


## References





## Additional context



Discord username (if different from GitHub): sidnov#8337

## Testing instructions


1. Spawn either on runway or C/D
2. Observe the chrono on the front panel
3. Use the in-sim weather panel to transition through different times of the day.
4. Observe brighter display in daylight and a dimmer display during night.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
